src/debuginfo.cpp

```diff
@@ -861,10 +880,22 @@ void jl_getFunctionInfo(char **name, size_t *line, char **filename,
         if (*line == (size_t) -1) {
             *line = prev.Loc.getLine();
         }
+
+        DILexicalBlockFile locscope = DILexicalBlockFile(prev.Loc.getScope((*it).second.func->getContext()));
+        jl_copy_str(filename, (char*)(uintptr_t)locscope.getFilename().data());
```

Comment:
Seems ok: http://stackoverflow.com/questions/2506793/c-life-span-of-temporary-arguments

also LLVM does it:
```
isaiah@titan:/cmn/jldev/deps/llvm-svn/lib$ grep -r "str().c_str" *
Analysis/Analysis.cpp:    *OutMessages = strdup(MsgsOS.str().c_str());
ExecutionEngine/ExecutionEngineBindings.cpp:                                       SectionName.str().c_str());
ExecutionEngine/ExecutionEngineBindings.cpp:                                       SectionName.str().c_str(),
LineEditor/LineEditor.cpp:  Data->EL = ::el_init(ProgName.str().c_str(), In, Out, Err);
LTO/LTOCodeGenerator.cpp:    CodegenOptions.push_back(strdup(o.first.str().c_str()));
Support/Unix/Path.inc:      ssize_t len = readlink(aPath.str().c_str(), exe_path, sizeof(exe_path));
Support/Unix/Program.inc:    int Err = posix_spawn(&PID, Program.str().c_str(), FileActions,
Support/Unix/Signals.inc:        OS << format("%s ", FunctionName.str().c_str());
Support/Unix/Signals.inc:        OS << format("%s", FileLineInfo.str().c_str());
Target/NVPTX/NVPTXReplaceImageHandles.cpp:    Idx = MFI->getImageHandleSymbolIndex(NewSymStr.str().c_str());
Target/NVPTX/NVPTXRegisterInfo.h:    return getStrPool()->getManagedString(O.str().c_str())->c_str();
Transforms/Scalar/SeparateConstOffsetFromGEP.cpp:        llvm_unreachable(RSO.str().c_str());
```